ci/github-script/reviewers.js

```diff
@@ -0,0 +1,129 @@
+
+  const teams = new Set(
+    owners
+      .map((handle) => handle.split('/'))
+      .filter(([org, slug]) => org === context.repo.owner && slug)
+      .map(([, slug]) => slug),
+  )
+  log('reviewers - teams', Array.from(teams).join(', '))
+
+  const team_members = new Set(
+    (await Promise.all(Array.from(teams, getTeamMembers)))
+      .flat(1)
+      .map(({ login }) => login),
+  )
+  log('reviewers - team_members', Array.from(team_members).join(', '))
+
+  const new_reviewers = users
+    .union(team_members)
+    // We can't request a review from the author.
+    .difference(new Set([pull_request.user?.login]))
+  log('reviewers - new_reviewers', Array.from(new_reviewers).join(', '))
+
+  // Filter users to repository collaborators. If they're not, they can't be requested
+  // for review. In that case, they probably missed their invite to the maintainers team.
+  const reviewers = (
+    await Promise.all(
+      Array.from(new_reviewers, async (username) => {
+        try {
+          await github.rest.repos.checkCollaborator({
+            ...context.repo,
+            username,
+          })
+          return username
+        } catch (e) {
+          if (e.status !== 404) throw e
+          log(
+            `User ${username} cannot be requested for review because they don't exist or are not a repository collaborator, ignoring. Probably missed the automated invite to the maintainers team (see <https://github.com/NixOS/nixpkgs/issues/234293>).`,
+          )
+        }
+      }),
+    )
+  ).filter(Boolean)
+  log('reviewers - reviewers', reviewers.join(', '))
+
+  if (reviewers.length > 15) {
+    log(
+      `Too many reviewers (${reviewers.join(', ')}), skipping review requests.`,
+    )
+    // false indicates, that we do have reviewers and don't need the "needs: reviewers" label.
+    return false
+  }
+
+  const requested_reviewers = new Set(
+    pull_request.requested_reviewers.map(({ login }) => login),
+  )
+  log(
+    'reviewers - requested_reviewers',
+    Array.from(requested_reviewers).join(', '),
+  )
+
+  const existing_reviewers = new Set(
+    reviews.map(({ user }) => user?.login).filter(Boolean),
+  )
+  log(
+    'reviewers - existing_reviewers',
+    Array.from(existing_reviewers).join(', '),
+  )
+
+  const non_requested_reviewers = new Set(reviewers)
+    .difference(requested_reviewers)
+    // We don't want to rerequest reviews from people who already reviewed.
+    .difference(existing_reviewers)
+  log(
+    'reviewers - non_requested_reviewers',
+    Array.from(non_requested_reviewers).join(', '),
+  )
+
+  if (non_requested_reviewers.size === 0) {
+    log('Has reviewer changes', 'false (skipped)')
+  } else if (dry) {
+    core.info(
+      `Requesting reviewers for #${pull_number}: ${Array.from(non_requested_reviewers).join(', ')} (dry)`,
+    )
+  } else {
+    await github.rest.pulls.requestReviewers({
+      ...context.repo,
+      pull_number,
+      reviewers,
+    })
+  }
```

Comment:
This is slightly different from the original script: if requesting a review from one user failed, the old script still requested reviews from all other users, whereas this will fail completely AFAIU.

See also your commit https://github.com/NixOS/nixpkgs/commit/034613f860fcd339bd2c20c8f6bc259a2f9dc034

Reply:
Judging by the commit message in https://github.com/NixOS/nixpkgs/commit/034613f860fcd339bd2c20c8f6bc259a2f9dc034, I guess this is actually somewhat important, since we did this switch once already.